lib/block_view_stake.go

```diff
@@ -1226,19 +1226,22 @@ func (bav *UtxoView) _connectStake(
 	// 4. Set the new ValidatorEntry.
 	bav._setValidatorEntryMappings(currentValidatorEntry)
 
-	// Increase the GlobalStakeAmountNanos.
-	// Retrieve the existing GlobalStakeAmountNanos.
-	// The PrevGlobalStakeAmountNanos will be restored if we disconnect this transaction.
-	prevGlobalStakeAmountNanos, err := bav.GetGlobalStakeAmountNanos()
-	if err != nil {
-		return 0, 0, nil, errors.Wrapf(err, "_connectStake: error retrieving GlobalStakeAmountNanos: ")
-	}
-	globalStakeAmountNanos, err := SafeUint256().Add(prevGlobalStakeAmountNanos, txMeta.StakeAmountNanos)
-	if err != nil {
-		return 0, 0, nil, errors.Wrapf(err, "_connectStake: error adding StakeAmountNanos to GlobalStakeAmountNanos: ")
+	// Increase the GlobalActiveStakeAmountNanos if the validator is active.
+	var prevGlobalActiveStakeAmountNanos *uint256.Int
+	if currentValidatorEntry.Status() == ValidatorStatusActive {
```

Comment:
I'm sure you've thougth about this, but do we want to make it so that you can't stake to a jailed validator? You would just add an `else if` block here to error if the status is jailed.

If you did this, I'd also add an `else` for "Unknown validator status" error.

Reply:
I thought about it more, and I think what you're doing here is more elegant. If we were to disallow staking to jailed validators, we'd break symmetry with Unstake, which I think adds complexity.